### PR TITLE
Fix `argmax`/`argmin` in CUB block reduction for F-order arrays with ndim > 1

### LINKED ISSUE #4061: argmax and argmin return wrong results for F-order arrays
On CuPy master:
```python
import cupy as cp
import numpy as np


cp.core.set_routine_accelerators([])
cp.core.set_reduction_accelerators([])

a = cp.asarray([[3, 1], [4, 2]], order='F')
b = cp.asnumpy(a)
print("cp.argmax:", a.argmax(), "  np.argmax:", b.argmax())
print("cp.argmin:", a.argmin(), "  np.argmin:", b.argmin())
```
output:
```
cp.argmax: 1   np.argmax: 2
cp.argmin: 2   np.argmin: 1
```
This bug affects both CuPy's original implementation as well as the CUB block reduction (in `cupy.core._cub_reduction`), but not the CUB device reduction (`cupy.cuda.cub`). My guess is for Fortran arrays we calculated the array index (`_J`) wrong...

### PR DESCRIPTION
**Pull request by leofang**
Fix #4061. ~~Blocked by #4062.~~ Parallel of #4084 to fix `argmax`/`argmin` when the CUB block reduction kernel is in use.

### CODE REVIEW

File: tests/cupy_tests/sorting_tests/test_search.py [hunk 1]
```diff
@@ -191,9 +199,20 @@ def test_cub_argmin(self, xp, dtype):
...
         ret = cupy.empty(())  # Cython checks return type, need to fool it
-        func = 'cupy.core._routines_statistics.cub.device_reduce'
-        with testing.AssertFunctionIsCalled(func, return_value=ret):
-            a.argmin()
+        if self.backend == 'device':
+            func_name = 'cupy.core._routines_statistics.cub.'
+            func_name += 'device_reduce'
+            with testing.AssertFunctionIsCalled(func_name, return_value=ret):
+                a.argmin()
+        elif self.backend == 'black':
```
Comment (grlee77) [inline tests/cupy_tests/sorting_tests/test_search.py:207]:
```suggestion
        elif self.backend == 'block':
```

File: tests/cupy_tests/sorting_tests/test_search.py [hunk 2]
```diff
@@ -211,9 +230,20 @@ def test_cub_argmax(self, xp, dtype):
...
         ret = cupy.empty(())  # Cython checks return type, need to fool it
-        func = 'cupy.core._routines_statistics.cub.device_reduce'
-        with testing.AssertFunctionIsCalled(func, return_value=ret):
-            a.argmax()
+        if self.backend == 'device':
+            func_name = 'cupy.core._routines_statistics.cub.'
+            func_name += 'device_reduce'
+            with testing.AssertFunctionIsCalled(func_name, return_value=ret):
+                a.argmax()
+        elif self.backend == 'black':
```
Comment (grlee77) [inline tests/cupy_tests/sorting_tests/test_search.py:238]:
```suggestion
        elif self.backend == 'block':
```

File: tests/cupy_tests/sorting_tests/test_search.py (hunk 1 above)
Comment (leofang) [reply tests/cupy_tests/sorting_tests/test_search.py:207]: Thanks! Another copy-paste error...

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
@asi1024 I merged with master. PTAL.

**Comment by asi1024** [comment]
LGTM!

**Pull request merged**